WordPress/WordPressUITests/Screens/SitePageScreen.swift

```diff
@@ -0,0 +1,135 @@
+        let app = XCUIApplication()
+        let createBlankPageButton = app.tables[ElementStringIDs.createBlankPageButton]
+        return createBlankPageButton.exists && createBlankPageButton.isHittable
+    }
+    // Check createBlankPageButton, closeIcon, blogButton elements to ensure Site Page is fully loaded
+    init() {
+        let app = XCUIApplication()
+        closeIcon = app.buttons[ElementStringIDs.closeIcon]
+        blogButton = app.textViews[ElementStringIDs.blogButton]
+        chooseALaytoutTitle = app.textViews[ElementStringIDs.chooseALaytoutTitle]
+        createBlankPageButton = app.buttons[ElementStringIDs.createBlankPageButton]
+        super.init(element: createBlankPageButton)
+    }
+
+    /**
+     Tap on "Create Blank Page button" to create a new blank page.
+     */
+    @discardableResult
+    func tapOnCreateBlankPageButton() -> SitePageScreen {
+        XCTAssert(createBlankPageButton.waitForExistence(timeout: 3))
+        XCTAssert(createBlankPageButton.waitForHittability(timeout: 3))
+        XCTAssert(createBlankPageButton.isHittable)
+        createBlankPageButton.tap()
+        return self
+    }
+
+    /**
+     Select the first available template to create a page from a layout.
+     */
+    func selectALayout() -> BlockEditorScreen {
+        selectedTemplate.tap()
+        XCTAssert(createPageButton.waitForExistence(timeout: 3))
+        createPageButton.tap()
+        return BlockEditorScreen()
+    }
+
+    /**
+     Tap on "Close" icon to close the activity.
+     */
+    func clickOnCloseIcon() {
+        XCTAssert(closeIcon.waitForExistence(timeout: 3))
+        XCTAssert(closeIcon.waitForHittability(timeout: 3))
+        XCTAssert(closeIcon.isHittable)
+        closeIcon.tap()
+    }
+
+    /**
+     Tap on "Publish" button to publish content.
+     */
+    func publish() -> EditorNoticeComponent {
+        XCTAssert(publishButton.waitForExistence(timeout: 3))
+        XCTAssert(publishButton.waitForHittability(timeout: 3))
+        publishButton.tap()
+        let secondPublishButton = XCUIApplication().descendants(matching: .button).containing(.button, identifier: "Publish").element(boundBy: 1)
+        secondPublishButton.tap()
+        return EditorNoticeComponent(withNotice: "Page published", andAction: "View")
+    }
+
+    /**
+     Open "Page Settings" page to custom content from EditorPostSettings.
+     */
+    func openPageSettings() -> EditorPostSettings {
+        moreButton.tap()
+        pageSettingsButton.tap()
+        return EditorPostSettings()
+    }
+
+    /**
+     Enters text into title field.
+     - Parameter text: the test to enter into the title
+     */
+    func enterTextInTitle(text: String) -> SitePageScreen {
+        if titleViewBlog.exists {
+            titleViewBlog.tap()
+            titleViewBlog.clearTextIfNeeded()
+            titleViewBlog.typeText(text)
+        } else {
+        titleView.tap()
+        titleView.typeText(text)
+        }
```

Comment:
The indentation in `else` blocks follow the same rules as for other blocks (true for several instances of `else` in your additions)

```suggestion
        } else {
            titleView.tap()
            titleView.typeText(text)
        }
```